packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -1121,108 +1162,88 @@ export function TransferPanel() {
       return true
     }
 
-    // Keep the button disabled while loading gas summary
-    if (
-      !ethBalance ||
-      (gasSummary.status !== 'success' && gasSummary.status !== 'unavailable')
-    ) {
-      return true
-    }
+    const ethBalanceFloat = isDepositMode
+      ? ethL1BalanceFloat
+      : ethL2BalanceFloat
 
-    return false
-  }, [
-    amountNum,
-    destinationAddressError,
-    isSmartContractWallet,
-    ethBalance,
-    requiredGasFees,
-    gasSummary.status,
-    isSwitchingL2Chain,
-    isTransferring,
-    selectedToken
-  ])
+    const selectedTokenBalanceFloat = isDepositMode
+      ? selectedTokenL1BalanceFloat
+      : selectedTokenL2BalanceFloat
 
-  const disableDeposit = useMemo(() => {
-    if (disableDepositAndWithdrawal) {
+    if (!ethBalanceFloat) {
       return true
     }
 
+    // Keep the button disabled while loading gas summary
     if (
-      selectedToken &&
-      isWithdrawOnlyToken(selectedToken.address, l2Network.id)
+      gasSummary.status !== 'success' &&
+      gasSummary.status !== 'unavailable'
```

Comment:
what about `gasSummary.status !== 'error'`?
or do we intend to disable transfer when gas summary throws an error?